tests/next_tests/unit_tests/iterator_tests/transforms_tests/test_cse.py

```diff
@@ -293,3 +299,27 @@ def test_scalar_extraction_inside_as_fieldop():
 
     actual = CSE.apply(testee, within_stencil=False)
     assert actual == expected
+
+
+def test_no_extraction_from_unapplied_lambda():
+    testee = im.if_(
+        "cond",
+        # the `+0`, and `+1` respectively, are only needed to avoid that f itself is collected
+        # which is not what we are interested in here
```

Comment:
Is this the comment in sync with the rest of the code? I don't understand what  `+0` and `+1` mean here. Maybe would help to copy here the IR/pseudocode example from the PR description...